BaseClasses.py

```diff
@@ -828,13 +828,41 @@ def get_connecting_entrance(self, is_main_entrance: typing.Callable[[Entrance],
         for entrance in self.entrances:  # BFS might be better here, trying DFS for now.
             return entrance.parent_region.get_connecting_entrance(is_main_entrance)
 
-    def add_locations(self, locations: Dict[str, Optional[int]], location_type: Optional[typing.Type[Location]] = None) -> None:
-        """Adds locations to the Region object, where location_type is your Location class and locations is a dict of
-        location names to address."""
+    def add_locations(self, locations: Dict[str, Optional[int]],
+                      location_type: Optional[typing.Type[Location]] = None) -> None:
+        """
+        Adds locations to the Region object, where location_type is your Location class and locations is a dict of
+        location names to address.
+        
+        :param locations: dictionary of locations to be created and added to this Region `{name: ID}`
+        :param location_type: Location class to be used to create the locations with"""
         if location_type is None:
             location_type = Location
         for location, address in locations.items():
             self.locations.append(location_type(self.player, location, address, self))
+    
+    def connect(self, connecting_region: Region, name: Optional[str] = None,
+                rule: Optional[Callable[[CollectionState], bool]] = None) -> None:
+        """
+        Connects this Region to another Region, placing the provided rule on the connection.
+        
+        :param connecting_region: Region object to connect to path is `self -> exiting_region`
+        :param name: name of the connection being created
+        :param rule: callable to determine access of this connection to go from self to the exiting_region"""
+        _exit = self.create_exit(name) if name else self.create_exit(f"{self.name} -> {connecting_region}")
+        if rule:
+            _exit.access_rule = rule
+        _exit.connect(connecting_region)
+    
+    def create_exit(self, name: str) -> Entrance:
+        """
+        Creates and returns an Entrance object as an exit of this region.
+        
+        :param name: name of the Entrance being created
+        """
+        _exit = Entrance(self.player, name, self)
```

Comment:
Instead of hard-coding `Entrance` here, we should add a `entrance_type: ClassVar[Type[Entrance]] = Entrance` to the class body and create the Entrance like this:
```suggestion
        _exit = self.entrance_tyoe(self.player, name, self)
```

Getting actual typing support for create_exit's return type would require a protocol or super class with typing.Generic, I think, but that is not strictly required since you can still override create_exit or cast and mypy will complain if `self.entrance_type` does not inherit from `Entrance`.